test: merge vmware function test in a single role

goneri opened the pull request:
##### SUMMARY

A lot of these roles should or already share some common operation. The
goal of this merge is to simplify the creation of composable scenario
depending on the test target.
e.g:
- only enable some tests on vcsim
- can validate this feature only on a physical lab
- use the same set of default variable
- share a setup.yml task between the different roles to initialize the
  environment
- use the same teardown task

This commit also disable 39 tests that are currently broken.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

vmware

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

goneri committed:
test: merge vmware function test in a single role

A lot of these roles should or already share some common operation. The
goal of this merge is to simplify the creation of composable scenario
depending on the test target.
e.g:
- only enable some tests on vcsim
- can validate this feature only on a physical lab
- use the same set of default variable
- share a setup.yml task between the different roles to initialize the
  environment
- use the same teardown task

This commit also disable 39 tests that are currently broken.

goneri committed:
test: teardown and setup playbook for vmware

Use two play called `teardown.yml` and `setup.yml` to handle the
teardown and prepare the environment. This allow to reduce the size
of the tests a lot.

goneri committed:
vmware: use generic variables instead of vcsim*

goneri committed:
vmware: use generic variables instead of vcsim*

Akasurde commented on test/integration/targets/vmware/tasks/setup.yaml:
```diff
@@ -0,0 +1,74 @@
+---
+- name: start vcsim
+  uri:
+    url: http://{{ vcsim }}:5000/spawn?cluster=2
```
We have some testcases which uses multiple objects.

```suggestion
    url: http://{{ vcsim }}:5000/spawn?datacenter=2&cluster=2&folder=2&ds=2&pool=2
```

Also, in future we need to add testcases for multiple object environment.

goneri replied:
I modified them a bit to avoid the constraint and they are all passing.

goneri replied:
to elaborate a bit, I would prefer to reduce the "default" environment because we need to re-prepare it before each test. This can quickly be really time consuming on real HW.
For the few cases where it matters, it's better to add the extra resources in the test itself.

goneri committed:
vmware_datastore_facts: get_all_object() fix for py3

With Python3, items() is an iterator that loop on all the key/val of the
dictionary. If the dict get changed during the iteration, Python will
raise the following error:
```
RuntimeError: dictionary changed size during iteration
```

This patch ensures we maniupulate a copy of the data structure.

goneri committed:
progress
